clients/drcachesim/tracer/instru.cpp

```diff
@@ -185,36 +185,30 @@ instru_t::instr_to_prefetch_type(instr_t *instr)
 bool
 instru_t::is_aarch64_icache_flush_op(instr_t *instr)
 {
-    if (instr_get_opcode(instr) != OP_sys)
-        return false;
-    switch (opnd_get_immed_int(instr_get_src(instr, 0))) {
+    switch (instr_get_opcode(instr)) {
     // TODO i#4406: Handle privileged icache operations.
-    case DR_IC_IVAU: return true;
+    case OP_ic_ivau: return true;
     }
     return false;
 }
 
 bool
 instru_t::is_aarch64_dcache_flush_op(instr_t *instr)
 {
-    if (instr_get_opcode(instr) != OP_sys)
-        return false;
-    switch (opnd_get_immed_int(instr_get_src(instr, 0))) {
+    switch (instr_get_opcode(instr)) {
     // TODO i#4406: Handle all privileged dcache operations.
-    case DR_DC_IVAC:
-    case DR_DC_CVAU:
-    case DR_DC_CIVAC:
-    case DR_DC_CVAC: return true;
+    case OP_dc_ivac:
+    case OP_dc_cvau:
+    case OP_dc_civac:
+    case OP_dc_cvac: return true;
     }
     return false;
 }
 
 bool
 instru_t::is_aarch64_dc_zva_instr(instr_t *instr)
 {
-    // TODO i#4393: Split OP_sys into multiple opcodes based on the operation.
-    return instr_get_opcode(instr) == OP_sys &&
-        opnd_get_immed_int(instr_get_src(instr, 0)) == DR_DC_ZVA;
+    return instr_get_opcode(instr) == OP_dc_zva;
```

Comment:
Please also remove the #4400 workaround from raw2trace.h.
However: we actually can't remove it nor the helper code like instr_summary_t::is_aarch64_dc_zva since we need to handle traces gathered with a tracer prior to this PR.
This PR needs to bump the offline raw trace version (the final trace_version_t is unchanged) OFFLINE_FILE_VERSION and add conditional use of the #4400 workaround for older versions.

Reply:
I don't think we can remove the #4400 workaround yet, as this PR doesn't back-align the dc zva address in any new way. When we do that, we should also add a test to verify back-alignment in online traces.

Reply:
To be resolved in post-v9.0 patch.